src/Dfc.CourseDirectory.Services/BulkUploadService/BulkUploadService.cs

```diff
@@ -356,56 +337,42 @@ public List<string> ProcessBulkUpload(Stream stream, int providerUKPRN, string u
             }
         }
 
-      
-
         public List<BulkUploadCourse> PolulateLARSData(List<BulkUploadCourse> bulkUploadcourses, out List<string> errors)
         {
             errors = new List<string>();
-            List<int> totalErrorList = new List<int>();
-            cachedQuals = new List<LarsSearchResultItem>();
+            var totalErrorList = new List<int>();
+            
+            var cachedQuals = new List<Lars>();
 
-            foreach (var bulkUploadcourse in bulkUploadcourses.Where(c => c.IsCourseHeader == true).ToList())
+            foreach (var bulkUploadcourse in bulkUploadcourses.Where(c => c.IsCourseHeader == true))
             {
-                LarsSearchCriteria criteria = new LarsSearchCriteria(bulkUploadcourse.LearnAimRef, 10, 0, string.Empty, null);
-                var qualifications = new List<LarsDataResultItem>();
-
-
-                var cachedResult = cachedQuals != null ? cachedQuals.FirstOrDefault(o => o.LearnAimRef == criteria.Search) : null;
+                var cachedResult = cachedQuals.FirstOrDefault(o => o.LearnAimRef == bulkUploadcourse.LearnAimRef);
+                
+                List<Lars> results = null;
 
-                List<LarsSearchResultItem> result = null;
                 if (cachedResult == null)
                 {
-                    result = Task.Run(async () => await _larsSearchService.SearchAsync(criteria)).Result.Value.Value.ToList();
-                    var qual = result.FirstOrDefault();
-                    if(qual != null)
+                    results = _larsSearchClient.Search(new LarsLearnAimRefSearchQuery
+                    {
+                        LearnAimRef = bulkUploadcourse.LearnAimRef,
+                        PageSize = 10
```

Comment:
```suggestion
                        PageSize = 1
```

Reply:
The original requested 10 and I didn't want to change the behaviour. However, one seems perfectly reasonable.